### PR TITLE
Add nebula-fade-poof SEXP

### LINKED ISSUE #2689: FR: Nebula rendering enhancements & configurability
As per discussion with The_E, Goober5000, & Axem -

- [ ] 1. Ability to fade nebula poof bitmaps in and out when toggled via sexp. Again, you can add or remove the bitmaps, but they immediately appear and disappear which is quite jarring when you're trying to do a natural transition.

- [x] 2. Ability to use an RGB color pick for nebula background instead of the presets. This one is probably less important, but would make everyone's lives easier and give people more freedom in color design for their nebula missions. Incidentally this would also circumvent the need for .pcx files as a color palette for the background.

- [x] 3. Ability to configure transparency, scale, rotation, density, velocity, etc of different nebula poofs individually via nebula.tbl. Right now, nebula textures need to be created a very certain way and match the background color in order to avoid looking too janky when fading in and out at the edges of the screen (which shouldn't even really be necessary). Ability to control these parameters per bitmap would allow a wider variety of art styles to be employed. As an example, [this](https://cdn.discordapp.com/attachments/433363226105741315/748326665670754354/unknown.png) may look great as a still image, but current rendering means these high quality poofs pop in and out of existence in an unrealistic looking fashion.

### PR DESCRIPTION
**Pull request by MjnMixael**
This fixes #2689 by adding a method to do the unfinished task.

`nebula-fade-poof` is different from `nebula-toggle-poof` in a major key way; it does not force a full regeneration of the poofs cloud. The two sexps could probably be merged, but I felt this was a simpler and cleaner approach. Fading poofs in and out really assumes the player continues to move around and offers a more smooth transition between available nebula types.

### CODE REVIEW

File: code/nebula/neb.h [hunk 1]
```diff
@@ -73,6 +73,10 @@ typedef struct poof_info {
 	::util::UniformFloatRange rotation;
 	float view_dist;
 	::util::UniformFloatRange alpha;
+	int fade_start;
+	int fade_duration;
+	bool fade_in;
+	float fade_multiplier;
```
Comment (Baezon) [inline code/nebula/neb.h:79]: The only thing I would say, is to separate these values from the others a bit and comment that they are dynamic values which can vary during a mission, unlike the others which are always completely static,

**Review by Baezon — CHANGES_REQUESTED** [review]
Good work, requiring the camera to move is unideal, but as you say, is much simpler and avoids any 'discontinutities'.

File: code/nebula/neb.h (hunk 1 above)
Comment (MjnMixael) [reply code/nebula/neb.h]: Good idea. Done.

**Review by Baezon — APPROVED** [review]

**Comment by MjnMixael** [comment]
Added a few fixes based on testing with Durandal

File: code/nebula/neb.cpp [hunk 2]
```diff
@@ -811,6 +871,17 @@ void upkeep_poofs()
...
+	if (!Neb2_poofs.empty()) {
+		for (size_t i = 0; i < Neb2_poofs.size();) {
+			if (!poof_is_used(Neb2_poofs[i].poof_info_index)) {
+				Neb2_poofs[i] = Neb2_poofs.back();
+				Neb2_poofs.pop_back();
+			} else // if we needed to cull we should not advance because we just moved a new poof into this spot
+				i++;
+		}
+	}
+
```
Comment (Baezon) [inline code/nebula/neb.cpp:884]: This should be moved into the above logic ( if too far || !is_used(..) cull it ). There can be an indefinite number of poofs so this is a potential performance concern. I *think* doing it that way is faster, but more importantly I think it's a bit clearer to have a single pass for culling.

**Pull request merged**